dataflow/java/src/main/java/com/httparchive/dataflow/BigQueryImport.java

```diff
@@ -288,20 +318,19 @@ public static void main(String[] args) {
                 DataflowPipelineWorkerPoolOptions.AutoscalingAlgorithmType.NONE);
 
         Pipeline p = Pipeline.create(pipelineOptions);
-
-        PCollectionTuple results = p.apply(TextIO.Read
-                .named("read-har")
-                .from(getHarBucket(options))
-                .withCompressionType(TextIO.CompressionType.GZIP)
-                .withCoder(HarJsonCoder.of()))
+        PCollectionTuple results = p
```

Comment:
The issue here is Java generics. Adding type annotations gets past the compile error you hit, and then you need a couple auxiliary transforms to massage the data into the right shape. Here's what I came up with.

```java
p.apply(Create.<GcsPath>of(getHarBucket(options)))
            .apply(ParDo.named("expand-glob").of(new ExpandGlobFn()))
            .apply(Flatten.<GcsPath>iterables())
            .apply(
                WithKeys.of(
                    new SerializableFunction<GcsPath, Integer>() {
                      @Override
                      public Integer apply(GcsPath path) {
                        return path.hashCode();
                      }
                    }))
            .apply(Reshuffle.<Integer, GcsPath>of())
            .apply(Values.<GcsPath>create())
            .apply(
                ParDo.named("split-har")
                    .withOutputTags(
                        BigQueryImport.PAGES_TAG,
                        TupleTagList.of(BigQueryImport.ENTRIES_TAG).and(BigQueryImport.BODIES_TAG))
                    .of(new DataExtractorFn()));
```

It is actually somewhat ironic that `Reshuffle` requires key-value pairs, because actually it should not be used in a key-sensitive way. If the hashcode is bad, this won't work well so you'll then want to switch to some other (preferably deterministic) way of assigning arbitrary keys for reshuffling.